src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -192,6 +192,11 @@ static ucs_status_t uct_rdmacm_cm_create_dummy_qp(struct rdma_cm_id *id,
     return UCS_OK;
 }
 
+static uint64_t uct_rdamcm_pd_get_key(const struct ibv_pd *pd)
+{
+    return (uintptr_t)pd ^ pd->handle;
```

Comment:
1. can you pls add comment about container?
2. maybe the key should be **both** pd and handle? and not a xor of them? there is some chance that pd1 and pd2 will be considered same because xor value would be the same. so need to make sure each of them is also compared.

Reply:
do you mean 86-bit key? maybe just set (or xor) the handle to hight 32 bits of 64-bit key initialized by the pd ptr?
```
return (uintptr_t)pd ^ ((uint64_t)pd->handle << 32);
```
or
```
uint64_t key64;
uint32_t *key32_p = &key64;

key32_p[0] = pd->handle;
key32_p[1] = (uint32_t)(uintptr_t)pd;
return key64;
```

Reply:
yes, i mean a struct of {pd, handle}
if we use xor, or reduce the pointer to 32 bit, it's not a 100% guarantee that we get the right PD, so could end up using CQ object with wrong PD

Reply:
just to clarify - should not `pd->handle` guarantee this in normal case? I cant find any mention in docs usage examples or purposes of this handle. If yes, then it does not makes sence to me otherwise we can even do simpler:
```
return pd->handle ? pd->handle : pd;
```

Reply:
use GUID as key:
`ibv_get_device_guid(struct ibv_device *device);`